fieldservice_maintenance/models/maintenance_request.py

```diff
@@ -19,12 +21,25 @@ def create(self, vals):
             fsm_equipment = self.env['fsm.equipment'].search(
                 [('maintenance_equipment_id', '=', request.equipment_id.id)],
                 limit=1)
-            fsm_order_id = self.env['fsm.order'].create(
-                {'type': 'maintenance',
-                 'equipment_id': fsm_equipment.id,
-                 'location_id': fsm_equipment and
-                 fsm_equipment.current_location_id.id,
-                 'request_id': request.id
-                 })
-            request.fsm_order_id = fsm_order_id
+            fsm_order_type = self.env['fsm.order.type'].search(
+                [('internal_type', '=', 'maintenance')],
+                order="id desc", limit=1)
+            try:
+                fsm_order_id = self.env['fsm.order'].create(
+                    {'type': fsm_order_type.id,
+                     'equipment_id': fsm_equipment.id,
+                     'location_id': fsm_equipment and
+                     fsm_equipment.current_location_id.id,
+                     'request_id': request.id
+                     })
+                request.fsm_order_id = fsm_order_id
+            except Exception:
+                if not fsm_equipment.current_location_id.id:
+                    raise UserError(_(
+                        'Missing current location on FSM equipment %s') % self.name)
+                else:
+                    raise UserError(_(
+                        'You cannot create the maintenance request. '
+                        'Please check the FSM equipment.'
+                    ))
```

Comment:
Nothing wrong in your code, but this is simpler and requires less code block indentation:
```suggestion
            if not fsm_equipment.current_location_id.id:
                raise UserError(_(
                    'Missing current location on FSM equipment %s') % self.name)
            fsm_order_id = self.env['fsm.order'].create(
                {'type': fsm_order_type.id,
                 'equipment_id': fsm_equipment.id,
                 'location_id': fsm_equipment.current_location_id.id,
                 'request_id': request.id
                 })
            request.fsm_order_id = fsm_order_id
```

Reply:
@dreispt very good.. thanks